model_analyzer/config/generate/base_model_config_generator.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .config_generator_interface import ConfigGeneratorInterface
+
+from model_analyzer.triton.model.model_config import ModelConfig
+import abc
+
+
+class BaseModelConfigGenerator(ConfigGeneratorInterface):
+    """ Base class for generating model configs """
+
+    # Dict of parameters to apply on top of the default config to result
+    # in the default config (none)
+    #
+    DEFAULT_PARAM_COMBO = {}
+
+    def __init__(self, config, model, client):
+        """
+        Parameters
+        ----------
+        config: ModelAnalyzerConfig
+        model: The model to generate ModelConfigs for
+        client: TritonClient
+        """
+        self._client = client
+        self._model_repository = config.model_repository
+        self._base_model = model
+        self._base_model_name = model.model_name()
+        self._cpu_only = model.cpu_only()
+        self._model_name_index = 0
+        self._live = False
+
+    def is_done(self):
+        """ Returns true if this generator is done generating configs """
+        return self._live and self._done_walking()
+
+    def next_config(self):
+        """
+        Returns
+        -------
+        ModelConfig
+            The next ModelConfig generated by this class
+        """
+        self._live = True
+        while True:
+            config = self._get_next_model_config()
+            yield (config)
```

Comment:
I don't understand how this doesn't return an instance count of 0 because we don't step before returning the first model config

Reply:
It worked correctly, but now is more explicit